library/std/src/sys/unix/time.rs

```diff
@@ -291,8 +291,13 @@ mod inner {
         pub fn now() -> Instant {
             #[cfg(target_os = "macos")]
             const clock_id: libc::clockid_t = libc::CLOCK_UPTIME_RAW;
-            #[cfg(not(target_os = "macos"))]
+
+            #[cfg(target_os = "linux")]
+            const clock_id: libc::clockid_t = libc::CLOCK_BOOTTIME;
```

Comment:
Should we have documentation saying that `Instant`, when possible, will use a clock that keeps going when the system is asleep -- and list on which targets this is definitely the case?

Reply:
When I fixed it for my crate I looked into most platforms: https://github.com/LiveSplit/livesplit-core/blob/092e336e24abd26e97e5a2e55abb0bbd97ff0b44/src/platform/normal/mod.rs#L6

Formatted as Markdown: https://gist.github.com/CryZe/afb052b82839d36d5bdbe4e88551f4dc

Reply:
Okay so it seems like after this PR it's pretty much "everything except macOS"?

Reply:
Fuchsia is also problematic as they specifically defined their monotonic to behave like Linux's... but they didn't provide an alternative BOOTTIME one yet. Also the PR only checks for "linux" specific and not Linux derivatives such as Android, which it probably should (just not emscripten as it errors out on BOOTTIME atm).

Reply:
Okay, doesn't look like this actually closes https://github.com/rust-lang/rust/issues/87906 quite yet then.